processing/src/main/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequence.java

```diff
@@ -828,6 +858,11 @@ public void offer(E item)
         throw new RuntimeException("Failed to offer result to output queue", e);
       }
     }
+
+    public void tryOffer(E item)
```

Comment:
Seems like an odd thing to ignore whether the offer succeeded or not.

If we think it should never fail, better to do something like `if (!this.queue.offer(item)) { throw defensive exception }` or even `if (!this.queue.offer(item)) { log a warning }`

If it might fail in normal operation, then what happens if it fails?

Reply:
it is slightly confusing, I agree. the only reason to write the 'terminal' signal is for cases where things fail early where something downstream is waiting on the blocking queue to have stuff in it before it has anything written to it, so that it can have something in it to stop waiting. If there is already stuff in the queue, or it is full, then writing the terminal marker doesn't matter because all of the things also check for the cancellation signal during processing. 

@kgyrtkirk was pushing for clearing out the queue so that we would always be able to write the terminal marker, but  something about that feels off to me, and it seems more expensive than just writing to queue if there is space and bailing, so I was trying to avoid doing that and came to this solution so we wouldn't block writing a terminal signal that we don't need if the queue has stuff in it.

Perhaps I should specialize this method into only being able to write the terminal signal? that way it won't accidentally be used when it does matter

Reply:
pushed up a change that turns `tryOffer` into `tryOfferTerminal` and added some comments to clarify the reason for it

Reply:
OK, thanks, makes sense to me.